app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User'
+
+  def self.update_post_counter(user_id)
+    user = User.find(user_id)
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
- [ Required ] I think it would be better if you used one of the [Active Rails callback](https://guides.rubyonrails.org/active_record_callbacks.html) e.g the `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method `private`. This ensures that the `posts_counter` field is incremented anytime a new post is added.